pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -406,25 +422,46 @@ func (p *ConfigurationEditFlags) Apply(
 		}
 	}
 
-	if cmd.Flags().Changed("annotation") {
-		annotationsMap, err := util.MapFromArrayAllowingSingles(p.Annotations, "=")
+	if cmd.Flags().Changed("annotation") || cmd.Flags().Changed("annotation-service") || cmd.Flags().Changed("annotation-revision") {
```

Comment:
Maybe, I'd prefer separate processing per changed flag, but let's wait for other opinions.

/cc @maximilien @rhuss @navidshaikh

Reply:
I have followed the pattern used for "label-service" and label-revision" for better consistency

Reply:
I'm fine with that (we can optimize later if we feel the need)